python/cugraph/centrality/betweenness_centrality.py

```diff
@@ -75,16 +106,52 @@ def betweenness_centrality(G, k=None, normalized=True,
     # workaround.
     #
     vertices = None
+    if implementation is None:
+        implementation = "default"
+    if implementation not in ["default", "gunrock"]:
+        raise Exception("Only two implementations are supported: 'default' "
+                        "and 'gunrock'")
+
     if k is not None:
-        raise Exception("sampling feature of betweenness "
-                        "centrality not currently supported")
+        if implementation == "gunrock":
+            raise Exception("sampling feature of betweenness "
+                            "centrality not currently supported "
+                            "with gunrock implementation, "
+                            "please use None or 'default'")
+        # In order to compare with preset sources,
+        # k can either be a list or an integer or None
+        #  int: Generate an random sample with k elements
+        # list: k become the length of the list and vertices become the content
+        # None: All the vertices are considered
+        if isinstance(k, int):
+            random.seed(seed)
+            vertices = random.sample(range(G.number_of_vertices()), k)
+        # Using k as a list allows to have an easier way to compare against
+        # other implementations on
+        elif isinstance(k, list):
+            vertices = k
+            k = len(vertices)
+        # FIXME: There might be a cleaner way to obtain the inverse mapping
+        if G.renumbered:
+            vertices = [G.edgelist.renumber_map[G.edgelist.renumber_map ==
+                                                vert].index[0] for vert in
+                        vertices]
+
+    if endpoints is not False:
+        raise NotImplementedError("endpoints accumulation for betweenness "
+                                  "centrality not currently supported")
 
     if weight is not None:
-        raise Exception("weighted implementation of betweenness "
-                        "centrality not currently supported")
+        raise NotImplementedError("weighted implementation of betweenness "
+                                  "centrality not currently supported")
+    if result_dtype not in [np.float32, np.float64]:
```

Comment:
Is there a better way or different types that should be accepted?

Reply:
What if we just return doubles? bc value often get over FLT MAX, correct?

Reply:
The Betweenness Centrality values themselves do not seem to get over FLT MAX and the GAP benchmark expects the result as:
> |V|-sized array of 32-bit floating point numbers
That is why I considered both float32 and float64 for the result type

Reply:
Moved the default value to float64 in 4bdf822 as it is less restrictive and kept the option to return float32